test/test_hyper.py

```diff
@@ -275,6 +276,18 @@ def test_closed_connections_are_reset(self):
         assert c.recent_stream is None
         assert c.next_stream_id == 1
         assert c.window_manager is not wm
+        with c._conn as conn:
+            assert conn.state_machine.state == ConnectionState.IDLE
+
+        c = HTTP20Connection('www.google.com')
```

Comment:
What I'm getting at is, assuming this patch changes something in the code, there should be some difference in the state of the connection between this point and the point after `close`. If there isn't, we're not testing the correct behaviour. ;)